src/connector/src/source/cdc/source/reader.rs

```diff
@@ -151,24 +151,33 @@ where
 
             let channel_ptr = Box::into_raw(tx) as i64;
```

Comment:
For current code, we don't even need wrapping the channel with box, since the `tx` variable is dropped after the method is called. We can pass `channel_ptr` as `&tx as *const Sender as i64`.